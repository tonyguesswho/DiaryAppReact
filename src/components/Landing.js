import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import logo from '../img/logo.png';
import Navbar from '../components/Navbar';


class Landing extends Component {
  render() {
    return (
      <header id="main-header">
        <img src={logo} alt={'logo'} className="logo" />
        <Navbar />
        <div className="home-text">
          <h1>Personalized Digital Diary</h1>
          <Link to="/create" className="btn btn-filled">
            Write
          </Link>
          <Link to="/entries" className="btn btn-transparent" id="myDiary">
            my Diary
          </Link>
        </div>
      </header>
    );
  }
}

export default Landing;
