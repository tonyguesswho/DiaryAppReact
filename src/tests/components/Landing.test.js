import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Landing from '../../components/Landing';

let wrapped;

beforeEach(() => {
  wrapped = shallow(<Landing />);
});

describe('Landing UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
