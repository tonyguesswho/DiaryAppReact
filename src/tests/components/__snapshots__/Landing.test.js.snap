// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing UI render features component should render as expected 1`] = `
<header
  id="main-header"
>
  <img
    alt="logo"
    className="logo"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg=="
  />
  <Navbar />
  <div
    className="home-text"
  >
    <h1>
      Personalized Digital Diary
    </h1>
    <Link
      className="btn btn-filled"
      replace={false}
      to="/create"
    >
      Write
    </Link>
    <Link
      className="btn btn-transparent"
      id="myDiary"
      replace={false}
      to="/entries"
    >
      my Diary
    </Link>
  </div>
</header>
`;
